scalafmt-core/shared/src/main/scala/org/scalafmt/internal/FormatOps.scala

```diff
@@ -1242,4 +1211,29 @@ class FormatOps(val tree: Tree, val initStyle: ScalafmtConfig) {
       case _ => (tree, select)
     }
 
+  @tailrec
+  final def findTokenWith[A](f: FormatToken => Option[A])(
+      ft: FormatToken,
+      step: Int,
+      end: Int = Int.MaxValue
+  ): Either[FormatToken, A] = {
+    f(ft) match {
+      case Some(a) => Right(a)
+      case _ =>
+        if (ft.right.end >= end) Left(ft)
+        else {
+          val nextFt = tokens(ft, step)
+          if (nextFt eq ft) Left(ft)
+          else findTokenWith(f)(nextFt, step, end)
+        }
+    }
+  }
+
+  final def findToken(
+      ft: FormatToken,
+      step: Int,
```

Comment:
I see that step always 1 or -1.

Maybe introduce adt instead of number constant?

```scala
sealed abstract class Step(raw: Int)
object Step {
  final case object Next extends Step(1)
  final case object Prev extends Step(-1)
}

Reply:
let me think about it.